app/views/sales/_action_buttons.html.haml

```diff
@@ -9,7 +9,7 @@
       - if @sale.can_edit_items?
         %button{:class=>'btn btn-primary', :ng_click=>"open($event, 'sm','salesContent' )"}
           = t(:ordered)
-        = render partial: 'shared/modal_state_change', locals: {idx: 'salesContent', model: @sale, form_path: state_change_sale_path(@sale), event: 'mark_prepared', return_path: sale_path(@sale)}
+        = render partial: 'shared/modal_state_change', locals: {idx: 'salesContent', model: @sale, form_path: state_change_sale_path(@sale), event: 'mark_prepared', return_path: sale_path(@sale), p2: :report_delivery}
```

Comment:
P2 is the variablename in the shared view. In this case it represent a checkbox for report_delivery.